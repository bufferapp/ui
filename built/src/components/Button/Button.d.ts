import React from 'react';
import PropTypes from 'prop-types';
export declare const ButtonWrapperStyled: any;
export declare const ButtonContainerStyled: any;
export declare const ButtonStyled: any;
declare const Button: {
    ({ disabled, onClick, type, size, label, isSplit, loading, icon, iconEnd, hasIconOnly, isSelect, items, selectPosition, onSelectClick, fullWidth, tooltip, ref, hideSearch, textToLeft, className, children, onOpen, ...props }: {
        [x: string]: any;
        disabled: any;
        onClick: any;
        type: any;
        size: any;
        label: any;
        isSplit: any;
        loading: any;
        icon: any;
        iconEnd: any;
        hasIconOnly: any;
        isSelect: any;
        items: any;
        selectPosition: any;
        onSelectClick: any;
        fullWidth: any;
        tooltip: any;
        ref: any;
        hideSearch: any;
        textToLeft: any;
        className: any;
        children: any;
        onOpen: any;
    }): React.JSX.Element;
    propTypes: {
        disabled: PropTypes.Requireable<boolean>;
        size: PropTypes.Requireable<string>;
        onClick: PropTypes.Validator<(...args: any[]) => any>;
        label: PropTypes.Requireable<string>;
        type: PropTypes.Requireable<string>;
        isSplit: PropTypes.Requireable<boolean>;
        isSelect: PropTypes.Requireable<boolean>;
        loading: PropTypes.Requireable<boolean>;
        hasIconOnly: PropTypes.Requireable<boolean>;
        icon: PropTypes.Requireable<PropTypes.ReactNodeLike>;
        iconEnd: PropTypes.Requireable<boolean>;
        items: PropTypes.Requireable<(PropTypes.InferProps<{
            id: PropTypes.Requireable<string>;
            title: PropTypes.Requireable<string>;
        }> | null | undefined)[]>;
        children: PropTypes.Requireable<PropTypes.ReactElementLike>;
        selectPosition: PropTypes.Requireable<string>;
        onSelectClick: PropTypes.Requireable<(...args: any[]) => any>;
        fullWidth: PropTypes.Requireable<boolean>;
        tooltip: PropTypes.Requireable<string>;
        ref: PropTypes.Requireable<PropTypes.ReactNodeLike>;
        hideSearch: PropTypes.Requireable<boolean>;
        className: PropTypes.Requireable<string>;
        onOpen: PropTypes.Requireable<(...args: any[]) => any>;
    };
    defaultProps: {
        disabled: boolean;
        isSplit: undefined;
        loading: boolean;
        size: string;
        type: string;
        label: undefined;
        hasIconOnly: boolean;
        icon: undefined;
        iconEnd: boolean;
        isSelect: undefined;
        items: undefined;
        selectPosition: string;
        onSelectClick: undefined;
        fullWidth: boolean;
        tooltip: undefined;
        ref: undefined;
        hideSearch: boolean;
        className: undefined;
        children: undefined;
        onOpen: null;
    };
};
export default Button;
//# sourceMappingURL=Button.d.ts.map