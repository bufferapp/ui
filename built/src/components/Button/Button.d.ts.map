{"version":3,"file":"Button.d.ts","sourceRoot":"","sources":["../../../../src/components/Button/Button.tsx"],"names":[],"mappings":"AAAA,OAAO,KAAK,MAAM,OAAO,CAAA;AACzB,OAAO,SAAS,MAAM,YAAY,CAAA;AAYlC,eAAO,MAAM,mBAAmB,KAM/B,CAAA;AAED,eAAO,MAAM,qBAAqB,KAEjC,CAAA;AAED,eAAO,MAAM,YAAY,KASxB,CAAA;AAoBD,QAAA,MAAM,MAAM;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;CA+GX,CAAA;AAsGD,eAAe,MAAM,CAAA"}