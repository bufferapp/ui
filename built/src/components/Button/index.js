export { default } from './Button';
//# sourceMappingURL=index.js.map