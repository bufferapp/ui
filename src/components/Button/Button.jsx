import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import * as Styles from './style';
import ChevronDown from '../Icon/Icons/ChevronDown';
import Select from '../Select/Select';

/*
Since buttons keep their own inline-block display type, we can only imitate this by using a wrapper (with
 `display: inline-block`) an internal container (with `display: flex`), and then the actual contents (including real
  buttons
 */
export const ButtonWrapperStyled = styled.div`
  ${Styles.ButtonWrapperBase};
  ${props => Styles[props.disabled ? 'disabled' : props.type]};
  ${props => (props.fullWidth ? Styles.fullWidth : '')};
`;

export const ButtonContainerStyled = styled.div`
  ${Styles.ButtonContainerBase};
`;

export const ButtonStyled = styled.button`
  ${Styles.ButtonNestedBase};
  ${props => Styles[props.size]};
  ${props => (props.disabled ? Styles.disabled : '')};
  ${props => (props.fullWidth ? Styles.fullWidth : '')};
  ${props => (props.loading && props.type === 'primary' ? Styles.primaryLoading : '')};
`;

const Loading = styled.img`
  width: 24px;
  margin-left: 10px;
`;

const VisuallyHiddenLabel = styled.span`
  position: absolute;
  overflow: hidden;
  clip: rect(0 0 0 0);
  height: 1px;
  width: 1px;
  margin: -1px;
  padding: 0;
  border: 0;
`;

/** All buttons, including text and split-buttons, follow the same core principles in dimensions, padding, and font sizes.
 * Combined with simple modifiers, they can be changed in size and appearance.  */
const Button = ({
  disabled,
  onClick,
  type,
  size,
  label,
  isSplit,
  loading,
  icon,
  iconEnd,
  hasIconOnly,
  isSelect,
  items,
  selectPosition,
  onSelectClick,
  fullWidth,
  tooltip,
  ref,
  hideSearch,
  className,
  children,
  ...props
}) =>  (
  <ButtonWrapperStyled
    className={className}
    disabled={disabled}
    type={type}
    fullWidth={fullWidth}
    $loading={loading}
  >
    <ButtonContainerStyled>
      <ButtonStyled
        onClick={!disabled ? onClick : undefined}
        disabled={disabled}
        isSplit={isSplit}
        icon={icon}
        hasIconOnly={hasIconOnly}
        data-tip={tooltip}
        ref={ref}
        aria-haspopup="false"
        size={size}
        fullWidth={fullWidth}
        loading={loading}
        type={type}
        {...props}
      >
        {!iconEnd && icon}
        {hasIconOnly && <VisuallyHiddenLabel>{label}</VisuallyHiddenLabel>}
        {!hasIconOnly && (
          <Styles.ButtonLabel hasIcon={!!icon} iconEnd={!!iconEnd}>
            {label}
          </Styles.ButtonLabel>
        )}
        {iconEnd && icon}

        {isSelect && (type === 'primary' || type === 'secondary') && (
          <Styles.ButtonArrow>
            <ChevronDown
              color={type === 'primary' ? 'white' : 'grayDark'}
              size={size}
              isChevron
            />
          </Styles.ButtonArrow>
        )}
        {loading && <Loading src="./images/loading-gray.svg" alt="loading" />}
      </ButtonStyled>
      {isSplit &&
        (type === 'primary' || type === 'secondary') &&
        (children ? (
          React.Children.map(children, child => React.cloneElement(child, {}))
        ) : (
          <Select
            onSelectClick={onSelectClick}
            items={items}
            type={type}
            isSplit
            yPosition={selectPosition}
            xPosition="right"
            disabled={disabled}
            hideSearch={hideSearch}
          />
        ))}
    </ButtonContainerStyled>
  </ButtonWrapperStyled>
  );

Button.propTypes = {
  /** Is the button disabled */
  disabled: PropTypes.bool,

  /** Size of the Button */
  size: PropTypes.oneOf(['small', 'large', 'medium']),

  /** OnClick handler */
  onClick: PropTypes.func.isRequired,

  /** Button label */
  label: PropTypes.string,

  /** Type of button */
  type: PropTypes.oneOf([
    'primary',
    'secondary',
    'text',
    'error',
    'danger',
  ]),

  /** Is the Button Split  */
  isSplit: PropTypes.bool,

  /** Is this the Select button with chevron */
  isSelect: PropTypes.bool,

  /** Is the Button Loading  */
  loading: PropTypes.bool,

  /** Does the button have only an icon and no label */
  hasIconOnly: PropTypes.bool,

  /** Icon to show with the label */
  icon: PropTypes.node,

  /** Icon to show with the label */
  iconEnd: PropTypes.bool,

  /** Items to display in the Split Button popup */
  items: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string,
      title: PropTypes.string,
    })
  ),

  /** Child element(s) to use for custom Split Button child content */
  children: PropTypes.element,

  /** Position of the Select popup */
  selectPosition: PropTypes.oneOf(['top', 'bottom']),

  /** Function to call on Split Button selected item click */
  onSelectClick: PropTypes.func,

  /** Is the button the full width of the parent container */
  fullWidth: PropTypes.bool,

  /** Tooltip to show on the component */
  tooltip: PropTypes.string,

  /** The prop to get the DOM element of the Button */
  ref: PropTypes.node,

  /** Is search hidden */
  hideSearch: PropTypes.bool,

  /** class passed by the dom element */
  className: PropTypes.string,
};

Button.defaultProps = {
  disabled: false,
  isSplit: undefined,
  loading: false,
  size: 'medium',
  type: 'secondary',
  label: undefined,
  hasIconOnly: false,
  icon: undefined,
  iconEnd: false,
  isSelect: undefined,
  items: undefined,
  selectPosition: 'bottom',
  onSelectClick: undefined,
  fullWidth: false,
  tooltip: undefined,
  ref: undefined,
  hideSearch: true,
  className: undefined,
  children: undefined,
};

export default Button;
