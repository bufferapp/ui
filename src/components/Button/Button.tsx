import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import * as Styles from './style'
import ChevronDown from '../Icon/Icons/ChevronDown'
import Select from '../Select/Select'

/*
Since buttons keep their own inline-block display type, we can only imitate this by using a wrapper (with
 `display: inline-block`) an internal container (with `display: flex`), and then the actual contents (including real
  buttons
 */

export const ButtonWrapperStyled = styled.div`
  ${Styles.ButtonWrapperBase};
  ${(props) =>
    // @ts-expect-error TS(7053) FIXME: Element implicitly has an 'any' type because expre... Remove this comment to see the full error message
    Styles[[props.type, props.disabled ? 'Disabled' : ''].join('')]};
  ${(props): string =>  
// @ts-expect-error TS(2322) FIXME: Type 'FlattenSimpleInterpolation | ""' is not assi... Remove this comment to see the full error message
 (props.fullWidth ? Styles.fullWidth : '')};
`

export const ButtonContainerStyled = styled.div`
  ${Styles.ButtonContainerBase};
`

export const ButtonStyled = styled.button`
  ${Styles.ButtonNestedBase};
  ${(props) =>
    // @ts-expect-error TS(7053) FIXME: Element implicitly has an 'any' type because expre... Remove this comment to see the full error message
    Styles[props.size]};
  ${(props) =>
    // @ts-expect-error TS(7053) FIXME: Element implicitly has an 'any' type because expre... Remove this comment to see the full error message
    Styles[props.disabled ? 'disabled' : '']};
  ${(props): string =>  
// @ts-expect-error TS(2322) FIXME: Type 'FlattenSimpleInterpolation | ""' is not assi... Remove this comment to see the full error message
 (props.fullWidth ? Styles.fullWidth : '')};
`

const Loading = styled.img`
  width: 24px;
  margin-left: 10px;
`

const VisuallyHiddenLabel = styled.span`
  position: absolute;
  overflow: hidden;
  clip: rect(0 0 0 0);
  height: 1px;
  width: 1px;
  margin: -1px;
  padding: 0;
  border: 0;
`

/** All buttons, including text and split-buttons, follow the same core principles in dimensions, padding, and font sizes.
 * Combined with simple modifiers, they can be changed in size and appearance.  */
const Button = ({
  // @ts-expect-error TS(7031) FIXME: Binding element 'disabled' implicitly has an 'any'... Remove this comment to see the full error message
  disabled,
  // @ts-expect-error TS(7031) FIXME: Binding element 'onClick' implicitly has an 'any' ... Remove this comment to see the full error message
  onClick,
  // @ts-expect-error TS(7031) FIXME: Binding element 'type' implicitly has an 'any' typ... Remove this comment to see the full error message
  type,
  // @ts-expect-error TS(7031) FIXME: Binding element 'size' implicitly has an 'any' typ... Remove this comment to see the full error message
  size,
  // @ts-expect-error TS(7031) FIXME: Binding element 'label' implicitly has an 'any' ty... Remove this comment to see the full error message
  label,
  // @ts-expect-error TS(7031) FIXME: Binding element 'isSplit' implicitly has an 'any' ... Remove this comment to see the full error message
  isSplit,
  // @ts-expect-error TS(7031) FIXME: Binding element 'loading' implicitly has an 'any' ... Remove this comment to see the full error message
  loading,
  // @ts-expect-error TS(7031) FIXME: Binding element 'icon' implicitly has an 'any' typ... Remove this comment to see the full error message
  icon,
  // @ts-expect-error TS(7031) FIXME: Binding element 'iconEnd' implicitly has an 'any' ... Remove this comment to see the full error message
  iconEnd,
  // @ts-expect-error TS(7031) FIXME: Binding element 'hasIconOnly' implicitly has an 'a... Remove this comment to see the full error message
  hasIconOnly,
  // @ts-expect-error TS(7031) FIXME: Binding element 'isSelect' implicitly has an 'any'... Remove this comment to see the full error message
  isSelect,
  // @ts-expect-error TS(7031) FIXME: Binding element 'items' implicitly has an 'any' ty... Remove this comment to see the full error message
  items,
  // @ts-expect-error TS(7031) FIXME: Binding element 'selectPosition' implicitly has an... Remove this comment to see the full error message
  selectPosition,
  // @ts-expect-error TS(7031) FIXME: Binding element 'onSelectClick' implicitly has an ... Remove this comment to see the full error message
  onSelectClick,
  // @ts-expect-error TS(7031) FIXME: Binding element 'fullWidth' implicitly has an 'any... Remove this comment to see the full error message
  fullWidth,
  // @ts-expect-error TS(7031) FIXME: Binding element 'tooltip' implicitly has an 'any' ... Remove this comment to see the full error message
  tooltip,
  // @ts-expect-error TS(7031) FIXME: Binding element 'ref' implicitly has an 'any' type... Remove this comment to see the full error message
  ref,
  // @ts-expect-error TS(7031) FIXME: Binding element 'hideSearch' implicitly has an 'an... Remove this comment to see the full error message
  hideSearch,
  // @ts-expect-error TS(7031) FIXME: Binding element 'textToLeft' implicitly has an 'an... Remove this comment to see the full error message
  textToLeft,
  // @ts-expect-error TS(7031) FIXME: Binding element 'className' implicitly has an 'any... Remove this comment to see the full error message
  className,
  // @ts-expect-error TS(7031) FIXME: Binding element 'children' implicitly has an 'any'... Remove this comment to see the full error message
  children,
  // @ts-expect-error TS(7031) FIXME: Binding element 'onOpen' implicitly has an 'any' t... Remove this comment to see the full error message
  onOpen,
  ...props
}) => (
  <ButtonWrapperStyled
    className={className}
    // @ts-expect-error TS(2769) FIXME: No overload matches this call.
    disabled={disabled}
    type={type}
    fullWidth={fullWidth}
    $loading={loading}
  >
    <ButtonContainerStyled>
      <ButtonStyled
        onClick={!disabled ? onClick : undefined}
        disabled={disabled}
        // @ts-expect-error TS(2769) FIXME: No overload matches this call.
        isSplit={isSplit}
        icon={icon}
        hasIconOnly={hasIconOnly}
        data-tip={tooltip}
        ref={ref}
        aria-haspopup="false"
        size={size}
        fullWidth={fullWidth}
        type={type}
        {...props}
      >
        {!iconEnd && icon}
        {hasIconOnly && <VisuallyHiddenLabel>{label}</VisuallyHiddenLabel>}
        {!hasIconOnly && (
          // @ts-expect-error TS(2769) FIXME: No overload matches this call.
          <Styles.ButtonLabel hasIcon={!!icon} iconEnd={!!iconEnd}>
            {label}
          </Styles.ButtonLabel>
        )}
        {iconEnd && icon}

        {isSelect && (type === 'primary' || type === 'secondary') && (
          // @ts-expect-error TS(2769) FIXME: No overload matches this call.
          <Styles.ButtonArrow textToLeft={textToLeft}>
            <ChevronDown
              color={type === 'primary' ? 'white' : 'grayDark'}
              size={size}
              isChevron
            />
          </Styles.ButtonArrow>
        )}

        {loading && <Loading src="./images/loading-gray.svg" alt="loading" />}
      </ButtonStyled>
      {isSplit &&
        (type === 'primary' || type === 'secondary') &&
        (children ? (
          React.Children.map(children, (child) => React.cloneElement(child, {}))
        ) : (
          <Select
            // @ts-expect-error TS(2769) FIXME: No overload matches this call.
            onSelectClick={onSelectClick}
            items={items}
            type={type}
            size={size}
            isSplit
            yPosition={selectPosition}
            xPosition="right"
            disabled={disabled}
            hideSearch={hideSearch}
            onOpen={onOpen}
          />
        ))}
    </ButtonContainerStyled>
  </ButtonWrapperStyled>
)

Button.propTypes = {
  /** Is the button disabled */
  disabled: PropTypes.bool,

  /** Size of the Button */
  size: PropTypes.oneOf(['small', 'large', 'medium']),

  /** OnClick handler */
  onClick: PropTypes.func.isRequired,

  /** Button label */
  label: PropTypes.string,

  /** Type of button */
  type: PropTypes.oneOf([
    'primary',
    'secondary',
    'text',
    'error',
    'danger',
    'orange',
  ]),

  /** Is the Button Split  */
  isSplit: PropTypes.bool,

  /** Is this the Select button with chevron */
  isSelect: PropTypes.bool,

  /** Is the Button Loading  */
  loading: PropTypes.bool,

  /** Does the button have only an icon and no label */
  hasIconOnly: PropTypes.bool,

  /** Icon to show with the label */
  icon: PropTypes.node,

  /** Icon to show with the label */
  iconEnd: PropTypes.bool,

  /** Items to display in the Split Button popup */
  items: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string,
      title: PropTypes.string,
    }),
  ),

  /** Child element(s) to use for custom Split Button child content */
  children: PropTypes.element,

  /** Position of the Select popup */
  selectPosition: PropTypes.oneOf(['top', 'bottom']),

  /** Function to call on Split Button selected item click */
  onSelectClick: PropTypes.func,

  /** Is the button the full width of the parent container */
  fullWidth: PropTypes.bool,

  /** Tooltip to show on the component */
  tooltip: PropTypes.string,

  /** The prop to get the DOM element of the Button */
  ref: PropTypes.node,

  /** Is search hidden */
  hideSearch: PropTypes.bool,

  /** class passed by the dom element */
  className: PropTypes.string,

  /** onOpen function to fire when the Dropdown menu is open */
  onOpen: PropTypes.func,
}

Button.defaultProps = {
  disabled: false,
  isSplit: undefined,
  loading: false,
  size: 'medium',
  type: 'secondary',
  label: undefined,
  hasIconOnly: false,
  icon: undefined,
  iconEnd: false,
  isSelect: undefined,
  items: undefined,
  selectPosition: 'bottom',
  onSelectClick: undefined,
  fullWidth: false,
  tooltip: undefined,
  ref: undefined,
  hideSearch: true,
  className: undefined,
  children: undefined,
  onOpen: null,
}

export default Button
