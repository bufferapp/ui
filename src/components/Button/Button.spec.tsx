/* eslint-disable react/jsx-filename-extension */
import React from 'react'
import snap from 'jest-auto-snapshots'
import renderer from 'react-test-renderer'
import Button from './Button'
import 'jest-styled-components'

snap(Button, './Button.jsx')

test('Button renders styles properly', () => {
  const tree = renderer
    .create(<Button type="primary" onClick={() => {}} label="Primary" />)
    .toJSON()
  expect(tree).toHaveStyleRule('color', '#FFFFFF')
  expect(tree).toHaveStyleRule('cursor', 'pointer')
})

test('Disabled button is disabled', () => {
  const tree = renderer
    .create(
      <Button type="primary" onClick={() => {}} label="Primary" disabled />,
    )
    .toJSON()
  expect(tree).toHaveStyleRule('background-color', '#c6cbe4')
  expect(tree).toHaveStyleRule('cursor', 'not-allowed')
})

test('Primary default loading button has a darker blue background', () => {
  const tree = renderer
    .create(
      <Button type="primary" onClick={() => {}} label="Primary" loading />,
    )
    .toJSON()
  expect(tree).toHaveStyleRule('background-color', '#121E66')
})
