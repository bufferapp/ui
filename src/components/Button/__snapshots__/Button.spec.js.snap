// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jest-auto-snapshots > Button Matches snapshot when array prop "items" has one item: "[object Object]" 1`] = `
<ForwardRef(styled.div)
  $loading={true}
  className="jest-auto-snapshots String Fixture"
  disabled={true}
  fullWidth={true}
  type="primary"
>
  <ForwardRef(styled.div)>
    <ForwardRef(styled.button)
      aria-haspopup="false"
      data-tip="jest-auto-snapshots String Fixture"
      disabled={true}
      fullWidth={true}
      hasIconOnly={true}
      icon={<NodeFixture />}
      isSplit={true}
      size="small"
    >
      <ForwardRef(styled.span)>
        jest-auto-snapshots String Fixture
      </ForwardRef(styled.span)>
      <NodeFixture />
      <ForwardRef(styled.div)>
        <ChevronDownIcon
          color="white"
          isChevron={true}
          size="small"
        />
      </ForwardRef(styled.div)>
      <ForwardRef(styled.img)
        alt="loading"
        src="./images/loading-gray.svg"
      />
    </ForwardRef(styled.button)>
    <NodeFixture />
  </ForwardRef(styled.div)>
</ForwardRef(styled.div)>
`;

exports[`jest-auto-snapshots > Button Matches snapshot when array prop "items" is an empty array: "[]" 1`] = `
<ForwardRef(styled.div)
  $loading={true}
  className="jest-auto-snapshots String Fixture"
  disabled={true}
  fullWidth={true}
  type="primary"
>
  <ForwardRef(styled.div)>
    <ForwardRef(styled.button)
      aria-haspopup="false"
      data-tip="jest-auto-snapshots String Fixture"
      disabled={true}
      fullWidth={true}
      hasIconOnly={true}
      icon={<NodeFixture />}
      isSplit={true}
      size="small"
    >
      <ForwardRef(styled.span)>
        jest-auto-snapshots String Fixture
      </ForwardRef(styled.span)>
      <NodeFixture />
      <ForwardRef(styled.div)>
        <ChevronDownIcon
          color="white"
          isChevron={true}
          size="small"
        />
      </ForwardRef(styled.div)>
      <ForwardRef(styled.img)
        alt="loading"
        src="./images/loading-gray.svg"
      />
    </ForwardRef(styled.button)>
    <NodeFixture />
  </ForwardRef(styled.div)>
</ForwardRef(styled.div)>
`;

exports[`jest-auto-snapshots > Button Matches snapshot when boolean prop "disabled" is set to: "false" 1`] = `
<ForwardRef(styled.div)
  $loading={true}
  className="jest-auto-snapshots String Fixture"
  disabled={false}
  fullWidth={true}
  type="primary"
>
  <ForwardRef(styled.div)>
    <ForwardRef(styled.button)
      aria-haspopup="false"
      data-tip="jest-auto-snapshots String Fixture"
      disabled={false}
      fullWidth={true}
      hasIconOnly={true}
      icon={<NodeFixture />}
      isSplit={true}
      onClick={[MockFunction]}
      size="small"
    >
      <ForwardRef(styled.span)>
        jest-auto-snapshots String Fixture
      </ForwardRef(styled.span)>
      <NodeFixture />
      <ForwardRef(styled.div)>
        <ChevronDownIcon
          color="white"
          isChevron={true}
          size="small"
        />
      </ForwardRef(styled.div)>
      <ForwardRef(styled.img)
        alt="loading"
        src="./images/loading-gray.svg"
      />
    </ForwardRef(styled.button)>
    <NodeFixture />
  </ForwardRef(styled.div)>
</ForwardRef(styled.div)>
`;

exports[`jest-auto-snapshots > Button Matches snapshot when boolean prop "fullWidth" is set to: "false" 1`] = `
<ForwardRef(styled.div)
  $loading={true}
  className="jest-auto-snapshots String Fixture"
  disabled={true}
  fullWidth={false}
  type="primary"
>
  <ForwardRef(styled.div)>
    <ForwardRef(styled.button)
      aria-haspopup="false"
      data-tip="jest-auto-snapshots String Fixture"
      disabled={true}
      fullWidth={false}
      hasIconOnly={true}
      icon={<NodeFixture />}
      isSplit={true}
      size="small"
    >
      <ForwardRef(styled.span)>
        jest-auto-snapshots String Fixture
      </ForwardRef(styled.span)>
      <NodeFixture />
      <ForwardRef(styled.div)>
        <ChevronDownIcon
          color="white"
          isChevron={true}
          size="small"
        />
      </ForwardRef(styled.div)>
      <ForwardRef(styled.img)
        alt="loading"
        src="./images/loading-gray.svg"
      />
    </ForwardRef(styled.button)>
    <NodeFixture />
  </ForwardRef(styled.div)>
</ForwardRef(styled.div)>
`;

exports[`jest-auto-snapshots > Button Matches snapshot when boolean prop "hasIconOnly" is set to: "false" 1`] = `
<ForwardRef(styled.div)
  $loading={true}
  className="jest-auto-snapshots String Fixture"
  disabled={true}
  fullWidth={true}
  type="primary"
>
  <ForwardRef(styled.div)>
    <ForwardRef(styled.button)
      aria-haspopup="false"
      data-tip="jest-auto-snapshots String Fixture"
      disabled={true}
      fullWidth={true}
      hasIconOnly={false}
      icon={<NodeFixture />}
      isSplit={true}
      size="small"
    >
      <ForwardRef(styled.div)
        hasIcon={true}
        iconEnd={true}
      >
        jest-auto-snapshots String Fixture
      </ForwardRef(styled.div)>
      <NodeFixture />
      <ForwardRef(styled.div)>
        <ChevronDownIcon
          color="white"
          isChevron={true}
          size="small"
        />
      </ForwardRef(styled.div)>
      <ForwardRef(styled.img)
        alt="loading"
        src="./images/loading-gray.svg"
      />
    </ForwardRef(styled.button)>
    <NodeFixture />
  </ForwardRef(styled.div)>
</ForwardRef(styled.div)>
`;

exports[`jest-auto-snapshots > Button Matches snapshot when boolean prop "hideSearch" is set to: "false" 1`] = `
<ForwardRef(styled.div)
  $loading={true}
  className="jest-auto-snapshots String Fixture"
  disabled={true}
  fullWidth={true}
  type="primary"
>
  <ForwardRef(styled.div)>
    <ForwardRef(styled.button)
      aria-haspopup="false"
      data-tip="jest-auto-snapshots String Fixture"
      disabled={true}
      fullWidth={true}
      hasIconOnly={true}
      icon={<NodeFixture />}
      isSplit={true}
      size="small"
    >
      <ForwardRef(styled.span)>
        jest-auto-snapshots String Fixture
      </ForwardRef(styled.span)>
      <NodeFixture />
      <ForwardRef(styled.div)>
        <ChevronDownIcon
          color="white"
          isChevron={true}
          size="small"
        />
      </ForwardRef(styled.div)>
      <ForwardRef(styled.img)
        alt="loading"
        src="./images/loading-gray.svg"
      />
    </ForwardRef(styled.button)>
    <NodeFixture />
  </ForwardRef(styled.div)>
</ForwardRef(styled.div)>
`;

exports[`jest-auto-snapshots > Button Matches snapshot when boolean prop "iconEnd" is set to: "false" 1`] = `
<ForwardRef(styled.div)
  $loading={true}
  className="jest-auto-snapshots String Fixture"
  disabled={true}
  fullWidth={true}
  type="primary"
>
  <ForwardRef(styled.div)>
    <ForwardRef(styled.button)
      aria-haspopup="false"
      data-tip="jest-auto-snapshots String Fixture"
      disabled={true}
      fullWidth={true}
      hasIconOnly={true}
      icon={<NodeFixture />}
      isSplit={true}
      size="small"
    >
      <NodeFixture />
      <ForwardRef(styled.span)>
        jest-auto-snapshots String Fixture
      </ForwardRef(styled.span)>
      <ForwardRef(styled.div)>
        <ChevronDownIcon
          color="white"
          isChevron={true}
          size="small"
        />
      </ForwardRef(styled.div)>
      <ForwardRef(styled.img)
        alt="loading"
        src="./images/loading-gray.svg"
      />
    </ForwardRef(styled.button)>
    <NodeFixture />
  </ForwardRef(styled.div)>
</ForwardRef(styled.div)>
`;

exports[`jest-auto-snapshots > Button Matches snapshot when boolean prop "isSelect" is set to: "false" 1`] = `
<ForwardRef(styled.div)
  $loading={true}
  className="jest-auto-snapshots String Fixture"
  disabled={true}
  fullWidth={true}
  type="primary"
>
  <ForwardRef(styled.div)>
    <ForwardRef(styled.button)
      aria-haspopup="false"
      data-tip="jest-auto-snapshots String Fixture"
      disabled={true}
      fullWidth={true}
      hasIconOnly={true}
      icon={<NodeFixture />}
      isSplit={true}
      size="small"
    >
      <ForwardRef(styled.span)>
        jest-auto-snapshots String Fixture
      </ForwardRef(styled.span)>
      <NodeFixture />
      <ForwardRef(styled.img)
        alt="loading"
        src="./images/loading-gray.svg"
      />
    </ForwardRef(styled.button)>
    <NodeFixture />
  </ForwardRef(styled.div)>
</ForwardRef(styled.div)>
`;

exports[`jest-auto-snapshots > Button Matches snapshot when boolean prop "isSplit" is set to: "false" 1`] = `
<ForwardRef(styled.div)
  $loading={true}
  className="jest-auto-snapshots String Fixture"
  disabled={true}
  fullWidth={true}
  type="primary"
>
  <ForwardRef(styled.div)>
    <ForwardRef(styled.button)
      aria-haspopup="false"
      data-tip="jest-auto-snapshots String Fixture"
      disabled={true}
      fullWidth={true}
      hasIconOnly={true}
      icon={<NodeFixture />}
      isSplit={false}
      size="small"
    >
      <ForwardRef(styled.span)>
        jest-auto-snapshots String Fixture
      </ForwardRef(styled.span)>
      <NodeFixture />
      <ForwardRef(styled.div)>
        <ChevronDownIcon
          color="white"
          isChevron={true}
          size="small"
        />
      </ForwardRef(styled.div)>
      <ForwardRef(styled.img)
        alt="loading"
        src="./images/loading-gray.svg"
      />
    </ForwardRef(styled.button)>
  </ForwardRef(styled.div)>
</ForwardRef(styled.div)>
`;

exports[`jest-auto-snapshots > Button Matches snapshot when boolean prop "loading" is set to: "false" 1`] = `
<ForwardRef(styled.div)
  $loading={false}
  className="jest-auto-snapshots String Fixture"
  disabled={true}
  fullWidth={true}
  type="primary"
>
  <ForwardRef(styled.div)>
    <ForwardRef(styled.button)
      aria-haspopup="false"
      data-tip="jest-auto-snapshots String Fixture"
      disabled={true}
      fullWidth={true}
      hasIconOnly={true}
      icon={<NodeFixture />}
      isSplit={true}
      size="small"
    >
      <ForwardRef(styled.span)>
        jest-auto-snapshots String Fixture
      </ForwardRef(styled.span)>
      <NodeFixture />
      <ForwardRef(styled.div)>
        <ChevronDownIcon
          color="white"
          isChevron={true}
          size="small"
        />
      </ForwardRef(styled.div)>
    </ForwardRef(styled.button)>
    <NodeFixture />
  </ForwardRef(styled.div)>
</ForwardRef(styled.div)>
`;

exports[`jest-auto-snapshots > Button Matches snapshot when passed all props 1`] = `
<ForwardRef(styled.div)
  $loading={true}
  className="jest-auto-snapshots String Fixture"
  disabled={true}
  fullWidth={true}
  type="primary"
>
  <ForwardRef(styled.div)>
    <ForwardRef(styled.button)
      aria-haspopup="false"
      data-tip="jest-auto-snapshots String Fixture"
      disabled={true}
      fullWidth={true}
      hasIconOnly={true}
      icon={<NodeFixture />}
      isSplit={true}
      size="small"
    >
      <ForwardRef(styled.span)>
        jest-auto-snapshots String Fixture
      </ForwardRef(styled.span)>
      <NodeFixture />
      <ForwardRef(styled.div)>
        <ChevronDownIcon
          color="white"
          isChevron={true}
          size="small"
        />
      </ForwardRef(styled.div)>
      <ForwardRef(styled.img)
        alt="loading"
        src="./images/loading-gray.svg"
      />
    </ForwardRef(styled.button)>
    <NodeFixture />
  </ForwardRef(styled.div)>
</ForwardRef(styled.div)>
`;

exports[`jest-auto-snapshots > Button Matches snapshot when passed only required props 1`] = `
<ForwardRef(styled.div)
  $loading={false}
  disabled={false}
  fullWidth={false}
  type="secondary"
>
  <ForwardRef(styled.div)>
    <ForwardRef(styled.button)
      aria-haspopup="false"
      disabled={false}
      fullWidth={false}
      hasIconOnly={false}
      onClick={[MockFunction]}
      size="medium"
    >
      <ForwardRef(styled.div)
        hasIcon={false}
        iconEnd={false}
      />
    </ForwardRef(styled.button)>
  </ForwardRef(styled.div)>
</ForwardRef(styled.div)>
`;
